### PR TITLE
Add argument to control MPC dedup prior to liftover

### PR DESCRIPTION
**Pull request by ch-kr**
Make default behavior not to deduplicate MPC prior to liftover, but maintain option to do so in the future

### CODE REVIEW

File: rmc/utils/mpc.py [hunk 1]
```diff
@@ -1052,9 +1055,19 @@ def liftover_mpc(
...
+        Default is False.
     :param overwrite_temp: Whether to overwrite intermediate HT with deduplicated MPC scores.
+        Only relevant if dedup_mpc is True.
         If False, will read from existing temporary path.
         Default is False.
+    :param failed_sites_str: Name of the field containing Boolean for whether site failed liftover.
+        Default is 'locus_fail_liftover'.
+    :param remove_liftover_annotations: Set of lifted over annotations to remove.
+        Default is `{"new_locus", "new_alleles"}` as these fields are the lifted over
+        versions of the original loci and alleles and are the new keys of the lifted over
```
Comment (lw453) [inline rmc/utils/mpc.py:1069]:
```suggestion
        versions of the original loci and alleles and are already copied to the new keys of the lifted over
```

File: rmc/utils/mpc.py [hunk 2]
```diff
@@ -1067,18 +1080,26 @@ def liftover_mpc(
...
-    )
-
     ht = default_lift_data(ht, remove_failed_sites=remove_failed_sites)
     # Drop unnecessary annotation if removing sites that failed liftover
     if remove_failed_sites:
-        ht = ht.drop("locus_fail_liftover")
+        ht = ht.drop(failed_sites_str)
+
+    # Drop additional unnecessary annotations if specified
+    # The defaults (`new_locus`, `new_alleles`) are already the new keys
```
Comment (lw453) [inline rmc/utils/mpc.py:1101]:
```suggestion
    # The defaults (`new_locus`, `new_alleles`) are already copied to the new keys
```

File: rmc/utils/mpc.py [hunk 3]
```diff
@@ -1040,7 +1040,10 @@ def liftover_mpc(
     freeze: int = CURRENT_FREEZE,
     gnomad_version: str = CURRENT_GNOMAD_VERSION,
     remove_failed_sites: bool = True,
+    dedup_mpc: bool = False,
     overwrite_temp: bool = False,
+    failed_sites_str: str = "locus_fail_liftover",
+    remove_liftover_annotations: Set[str] = {"new_locus", "new_alleles"},
```
Comment (lw453) [inline rmc/utils/mpc.py:1046]: Just want to double check, there's also `original_locus`, `original_alleles`, and `ref_allele_mismatch`. We're leaving those in there? (Can't remember what we did for release)

Comment (ch-kr) [reply rmc/utils/mpc.py:1046]: yep we left those but can remove
```
hailctl describe gs://regional_missense_constraint/MPC/2.1.1/7/mpc_liftover38_dedup.ht
----------------------------------------
File Type: Table
    Partitions: 9761
    Rows: 67996604
    Empty partitions: 0
    Min(rows/partition): 629
    Max(rows/partition): 165302
    Median(rows/partition): 6251
    Mean(rows/partition): 6966
    StdDev(rows/partition): 3899
----------------------------------------
Global fields:

----------------------------------------
Row fields:
    'locus': locus<grch38>
    'alleles': array<str>
    'mpc': float64
    'transcript_grch37': str
    'original_locus': locus<grch37>
    'original_alleles': array<str>
    'ref_allele_mismatch': bool
----------------------------------------
Key: ['locus', 'alleles']
----------------------------------------
```

**Review by lw453 — APPROVED** [review]

**Pull request merged**